sg_jira/handlers/entity_issue_handler.py

```diff
@@ -903,20 +923,8 @@ def _get_shotgun_assignment_from_jira_issue_change(
         to_assignee = change["to"]
         if data_type == "multi_entity":
             if from_assignee:
-                # Try to remove the old assignee from the Shotgun assignment
-                jira_user = self._jira.user(from_assignee)
-                sg_user = self._shotgun.find_one(
-                    "HumanUser",
-                    [["email", "is", jira_user.emailAddress]],
-                    ["email", "name"]
-                )
-                if not sg_user:
-                    self._logger.debug(
-                        "Unable to find a Shotgun user with email address %s" % (
-                            jira_user.emailAddress
-                        )
-                    )
-                else:
+                sg_user = self._jira_user_to_shotgun(shotgun_field, user_id=from_assignee, raise_on_missing_user=False)
```

Comment:
The code to search for a user was repeated ad-nauseam, so I refactored it, especially since I needed it to change based on a JIRA server or JIRA cloud.

Reply:
You'll notice that the logic is always the same
When looking at the `from` field  (`if from_assignee`), we're passing in only the id. If the search fails, we simply log an error